test/llvm/llvm-invariant/function_local_const.chpl

```diff
@@ -0,0 +1,21 @@
+// This test checks if llvm.invariant.start is added to function local constants
+// Normally primitive type constants could be used, but chapel compiler
+// performs some constant propagation, which doesn't happen in case of records.
+
+record A
+{
+  var a : int;
+}
```

Comment:
We have some ongoing work with initializers. Can you add a local and global test using this pattern?

``` chapel
record A
{
  var a : int;
  proc init(arg:int) {
    a = arg;
    super.init();
  }
}
```

That will cause the compiler to follow some code paths that aren't exercised in your tests so far but which we expect to become widely usedin the future.